src/dolphin/manager.ts

```diff
@@ -30,7 +30,7 @@ export class DolphinManager {
     const dolphinInstall = this.getInstallation(dolphinType);
     await dolphinInstall.validate({
       onProgress: (current, total) => this._onProgress(dolphinType, current, total),
-      onComplete: () => this._onComplete(dolphinType),
+      onComplete: async () => this._onComplete(dolphinType, await dolphinInstall.getDolphinVersion()),
```

Comment:
```suggestion
      onComplete: () => dolphinInstall.getDolphinVersion().then(version => {
        this._onComplete(dolphinType, version); 
      }),
```

I think I would prefer the callback to be synchronous.